tests/helpers/globify.ts

```diff
@@ -0,0 +1,7 @@
+export type Globify<S extends string> = S extends `${infer Head}_${infer Tail}`
```

Comment:
Could you add a short comment explaining what this expression is doing, since it's not particularly easy to parse?

Reply:
```suggestion
// takes a string like `foo_bar`, recursively splits it up along underscores, and collects all kinds of glob expressions.
// for `foo_bar`, it'd evaluate to `'*' | 'foo*' | 'foo_*' | 'foo_bar'`.
export type Globify<S extends string> = S extends `${infer Head}_${infer Tail}`
```